lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return array1 == nil && array2 == nil
+  elsif array1.length != array2.length
+    return false
+  elsif array1.length == 0 || array2.length == 0
+    return array1.length == array2.length
```

Comment:
Same feedback as on line 6, it's easier to read and maintain if you return a clear boolean rather than nested logic.